base/unicode/utf8.jl

```diff
@@ -312,6 +312,11 @@ function convert(::Type{UTF8String}, a::Vector{UInt8}, invalids_as::AbstractStri
 end
 convert(::Type{UTF8String}, s::AbstractString) = utf8(bytestring(s))
 
+function convert{T<:Union{UInt16, UInt32}}(::Type{UTF8String}, a::Vector{T}, len=length(a)-1)
+    encode_to_utf8(T, a, len)
```

Comment:
It's a little weird that we have to pass `T` to `encode_to_utf8`.    Couldn't we define `encode_to_utf8{T}(a::Vector{T}, len)`?   Is there any case where we call `encode_to_utf8(T, a, len)` in which `a` is not a `Vector{T}`?